hera_cal/delay_filter.py

```diff
@@ -123,21 +138,95 @@ def load_delay_filter_and_write(infilename, calfile=None, Nbls_per_load=None, sp
             df.hd.data_array = None  # this forces a reload in the next loop
 
+
+def load_delay_filter_and_write_baseline_list(datafile_list, baseline_list, calfile_list=None, spw_range=None, cache_dir=None,
+                                              read_cache=False, write_cache=False, round_up_bllens=False,
+                                              factorize_flags=False, time_thresh=0.05,
+                                              res_outfilename=None, CLEAN_outfilename=None, filled_outfilename=None,
+                                              clobber=False, add_to_history='', **filter_kwargs):
+    '''
+    Uses partial data loading and writing to perform delay filtering.
+
+    Arguments:
+        datafile_list: list of data files to perform cross-talk filtering on
+        baseline_list: list of antenna-pair-pol triplets to filter and write out from the datafile_list.
+        calfile_list: optional list of calibration files to apply to data before xtalk filtering
+        spw_range: 2-tuple or 2-list, spw_range of data to filter.
+        cache_dir: string, optional, path to cache file that contains pre-computed dayenu matrices.
+                    see uvtools.dspec.dayenu_filter for key formats.
+        read_cache: bool, If true, read existing cache files in cache_dir before running.
+        write_cache: bool. If true, create new cache file with precomputed matrices
+                           that were not in previously loaded cache files.
+        round_up_bllens: bool, if True, round up baseline lengths. Default is False.
+        factorize_flags: bool, optional
+            If True, factorize flags before running delay filter. See vis_clean.factorize_flags.
+        time_thresh : float
+            Fractional threshold of flagged pixels across time needed to flag all times
+            per freq channel. It is not recommend to set this greater than 0.5.
+            Fully flagged integrations do not count towards triggering time_thresh.
+        res_outfilename: path for writing the filtered visibilities with flags
+        CLEAN_outfilename: path for writing the CLEAN model visibilities (with the same flags)
+        filled_outfilename: path for writing the original data but with flags unflagged and replaced
+            with CLEAN models wherever possible
+        clobber: if True, overwrites existing file at the outfilename
+        add_to_history: string appended to the history of the output file
+        filter_kwargs: additional keyword arguments to be passed to DelayFilter.run_delay_filter()
+    '''
+    hd = io.HERAData(datafile_list, filetype='uvh5')
+    if spw_range is None:
+        spw_range = [0, hd.Nfreqs]
+    freqs = hd.freq_array.flatten()[spw_range[0]:spw_range[1]]
+    baseline_antennas = []
+    for blpolpair in baseline_list:
+        baseline_antennas += list(blpolpair[:2])
+    baseline_antennas = np.unique(baseline_antennas).astype(int)
+    if calfile_list is not None:
```

Comment:
This block of code is repeated in several places and should be broken out into its own function.

Reply:
I've broken this out into `io.initialize_calfits_from_list`